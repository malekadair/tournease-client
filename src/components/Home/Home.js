import React, { Component } from "react";
import TournamentList from "../TournamentList/TournamentList";
import MainContent from "../MainContent/MainContent";
import TokenService from "../../services/TokenService";

class Home extends Component {
  renderTournamentList = () => {
    return (
      <div>
        <h2 id="upcoming">Upcoming Tournaments:</h2>
        <TournamentList />
      </div>
    );
  };
  renderLogin = () => {
    return <MainContent />;
  };
  render() {
    return (
      <div className="home">
        {TokenService.hasAuthToken()
          ? this.renderTournamentList()
          : this.renderLogin()}
      </div>
    );
  }
}

export default Home;
