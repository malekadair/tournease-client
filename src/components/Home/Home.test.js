import React from "react";
import toJson from "enzyme-to-json";
import Home from "./Home";

describe(`Home component`, () => {
  const props = {
    id: "a",
    name: "test-class-name"
  };

  it("renders a .Home by default", () => {
    const wrapper = <Home />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders the Home given props", () => {
    const wrapper = <Home {...props} />;
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
