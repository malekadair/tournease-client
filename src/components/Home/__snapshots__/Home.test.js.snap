// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders a .Home by default 1`] = `undefined`;

exports[`Home component renders the Home given props 1`] = `undefined`;
